### DISCUSSION TITLE
Warning: Text content did not match. Server: "Injected (unsupported)" Client: "MetaMask"

### DISCUSSION THREAD

**Discussion by ranlix** [wevm/wagmi#46] [General]
When I clone examples/next to local, visit index page, found that metamask button is disabled, it seems that metamask in connectors is not ready, and there is another warning says 
`Warning: Text content did not match. Server: "Injected" Client: "MetaMask"
    at button`
![image](https://user-images.githubusercontent.com/4019372/148580659-8c4ae8a3-4048-4d3f-a818-8e2581ef43be.png)

So, what's the reason leads that issue?

**Comment by jem-computer** [comment]
See https://github.com/tmm/wagmi/issues/28 for a solution - example needs to be fixed :)

**Reply by iduuck** [reply]
I would not suggest doing so, since (at least for the example) this is a crucial element, which shouldn't be delayed (or dynamically loaded). I would better suggest, that you are not relying on the `name` of the connector until we have fixed the problem internally.

**Reply by ranlix** [reply]
I have used that "useIsMounted" to work around it, looking forward to the new fix

**Reply by iduuck** [reply]
Actually you might already be good to go right now without the fix. You just need to **not** rely on the name attribute. I will come up with an example for this soon!

@liran319 Would you mind opening an issue for this?

**Reply by ranlix** [reply]
No.
Here is the reported issue: https://github.com/tmm/wagmi/issues/51
